lib/index.js

```diff
@@ -25,31 +26,56 @@ internals.store.lookUp = function (obj, path) {
   };
 };
 
-exports.replace = function (obj, path, fn) {
+exports.replace = function (obj, path, fn, options) {
   Assert.ok(obj, 'obj must be defined');
   Assert.ok(typeof obj === 'object' || typeof obj === 'function', 'obj must be an object or a constructor');
   var func = Get(obj, path);
   Assert.strictEqual(typeof func, 'function', 'path must be a valid function of obj');
   Assert.strictEqual(typeof fn, 'function', 'fn must be a function object');
 
+  var find = internals.store.lookUp(obj, path);
+  Assert.strictEqual(find.result, undefined, 'there is already a replace for "obj"[' + path + ']');
+
+  options = Merge({}, options);
+
+  var invocations = 0;
   var stand = {
     obj: obj,
     path: path
   };
-  var find = internals.store.lookUp(obj, path);
-  Assert.strictEqual(find.result, undefined, 'there is already a replace for "obj"[' + path + ']');
   internals.store.push(stand);
 
+  function restore () {
+    var find = internals.store.lookUp(obj, path);
+    internals.store.splice(find.index, 1);
+    Set(obj, path, this.original);
+  }
+
   var result = {
     original: func,
-    restore: function () {
-      var find = internals.store.lookUp(obj, path);
-      internals.store.splice(find.index, 1);
-      Set(obj, path, this.original);
-    },
+    restore: restore,
+    get invocations () { return invocations; },
     _stand: stand
   };
 
-  Set(obj, path, fn.bind(obj, result));
+  function wrapper () {
+    invocations++;
+
+    if (invocations === options.restoreAfter) {
```

Comment:
I don't like `ttl`. It implies a passage of an amount of time, which is completely unrelated to this. I'd like to use a name that has a well defined meaning or something that actually describes what it does. I couldn't come up with a well defined term.